src/judoscale/django/middleware.py

```diff
@@ -28,5 +29,16 @@ def __call__(self, request):
 
             metrics_store.add(metric)
             logger.debug("queue_time={}ms".format(round(queue_time_ms, 2)))
+            self._get_reporter(reporter)
 
         return self.get_response(request)
+
+    def _get_reporter(self, reporter):
+        try:
+            if not reporter._thread.is_alive():
+                import os
+                pid = os.getpid()
```

Comment:
Is `pid` being used anywhere?

Reply:
Yes in the [_build_reporter](https://github.com/judoscale/judoscale-python/blob/ed472cb72e2931cd237414988b70c56851114424/src/judoscale/core/reporter.py#L41) method, and it reflects the pid of the worker/process we see when the app starts

> 3:41:33 PM web.1   |  INFO - [Judoscale] **Starting reporter for process 27701**
> 3:41:33 PM web.1   |  DEBUG - [Judoscale] **Posting metrics {'dyno': 'web.1', 'pid': 27701**, 'config': {'log_level': 'DEBUG', 'report_interval_seconds': 2}, 'metrics': []} to Judoscale adapter API https://judoscale-python.requestcatcher.com/v1/metrics
> 3:41:33 PM web.1   |  [2022-07-22 12:41:33 +0000] [27701] [INFO] Starting gunicorn 20.1.0
> 3:41:33 PM web.1   |  [2022-07-22 12:41:33 +0000] [27701] [INFO] Listening at: http://0.0.0.0:5100 (27701)
> 3:41:33 PM web.1   |  [2022-07-22 12:41:33 +0000] [27701] [INFO] Using worker: sync
> 3:41:33 PM web.1   |  [2022-07-22 12:41:33 +0000] [27752] [INFO] **Booting worker with pid: 27752**
> 3:41:34 PM proxy.1 |  Listening on port 5000
> 3:41:34 PM proxy.1 |  Proxying requests to port 5100
> ...
> 3:41:51 PM web.1   |  DEBUG - [Judoscale] queue_time=7.29ms
> 3:41:51 PM web.1   |  INFO - [Judoscale] **Starting reporter for process 27752** (here it is the new reporter called from within the middleware when we hit the hello world page)
> 3:41:51 PM web.1   |  DEBUG - [Judoscale] **Posting metrics {'dyno': 'web.1', 'pid': 27752**, 'config': {'log_level': 'DEBUG', 'report_interval_seconds': 2}, 'metrics': [[1658493712, 7.29, 'queue_time', None]]} to Judoscale adapter API https://judoscale-python.requestcatcher.com/v1/metrics
> 3:41:51 PM web.1   |  Hello, world

Reply:
I see. Is there any reason to call `os.getpid()` here, though? Looks like we're not using this `pid` local variable.

Reply:
Oh right, my bad, I moved the pid in the log of the reporter.start() but forgot to delete it here.